### PR TITLE
1482 prepare instructions for upgrading ima on schain side

### LINKED ISSUE #188: Prepare instructions for upgrade
Need to prepare instruction for chain owners how to upgrade predeployed contracts using these tools

### LINKED ISSUE #1482: Prepare instructions for upgrading IMA on schain side

### PR DESCRIPTION
**Pull request by yavrsky**

### CODE REVIEW

File: proxy/scripts/magic_upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -e
+
```
Comment (DimaStebaev) [inline proxy/scripts/magic_upgrade.sh:4]: Missing check that all necessary environment variables are passed.

File: proxy/scripts/magic_upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+rm -f ima-$DEPLOYED_VERSION-predeployed-abi.json
+wget $IMA_RELEASES_URL/$DEPLOYED_VERSION/ima-$DEPLOYED_VERSION-predeployed-abi.json
+cd ../scripts/
+wget $IMA_RELEASES_URL/$DEPLOYED_VERSION/ima-schain-$DEPLOYED_VERSION-manifest.json
+python3 change_manifest.py ima-schain-$DEPLOYED_VERSION-manifest.json
+mv ima-schain-$DEPLOYED_VERSION-manifest.json ../.openzeppelin/unknown-$SCHAIN_ID.json
+cd ..
+
+git clone https://github.com/skalenetwork/skale-network.git
+LATEST_STABLE_IMA_VERSION=$(ls skale-network/releases/mainnet/IMA/ | sort -r | head -n 1)
```
Comment (DimaStebaev) [inline proxy/scripts/magic_upgrade.sh]: It's not always possible to upgrade to the latest version.

File: proxy/scripts/magic_upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+#sed version inside upgrade script
+
+SCHAIN_NAME=$SCHAIN_NAME \
+ABI="data/ima-$DEPLOYED_VERSION-predeployed-abi.json" \
+IMA_ABI="data/ima-$LATEST_STABLE_IMA_VERSION-mainnet-abi.json" \
+MAINNET_CHAIN_ID=$MAINNET_CHAIN_ID \
+SAFE_ADDRESS=$SAFE_ADDRESS \
+ALLOW_NOT_ATOMIC_UPGRADE="OK" \
```
Comment (DimaStebaev) [inline proxy/scripts/magic_upgrade.sh]: It depends on the upgrade. It's not OK in general case.

File: proxy/scripts/upgrade_instruction.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Step-by-step guide for upgrading IMA contracts on Schain side
+
+## Installing project
+Git clone and run yarn install in the root of the project.
+## Preparing environment
```
Comment (DimaStebaev) [inline proxy/scripts/upgrade_instruction.md]: It's good to provide an example.

File: proxy/scripts/upgrade_instruction.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+First of all you need to create new account. You can generate it in Metamask. Next export private key and put it in `.env` as `PRIVATE_KEY` without 0x by path `IMA/proxy/`. Also put endpoint of your chain as `ENDPOINT` to the `.env`. Also you need to add this account to your Gnosis Wallet as one of the Safe owners. Open Gnosis app, go to "Settings", click "Add new owner" and then sign transaction.
+## Granting role
+Next you need to grant `DEPLOYER_ROLE` for account that we have created in previous step for being able to deploy new contracts on chain. One of the options you can use multisigwallet-cli to encode the transaction and send it via IMA to the Skale chain. Also you can ask somebody from Skale engineering team to do it for you.
```
Comment (DimaStebaev) [inline proxy/scripts/upgrade_instruction.md]: I'm not sure that SKALE team has permissions to do it.

**Review by DimaStebaev — DISMISSED** [review]

**Review by DimaStebaev — DISMISSED** [review]

File: proxy/scripts/upgrade_instruction.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+```
+## Prepare environment
+ Create `.env` file by path `IMA/proxy/`. Now you need to create new account. You can generate it in Metamask. Next export private key and put it in `.env` as `PRIVATE_KEY` without 0x. Also put endpoint of your chain as `ENDPOINT` to the `.env`. Also you need to add this account to your Gnosis Wallet as one of the Safe owners. Open Gnosis app, go to "Settings", click "Add new owner" and then sign transaction. Example of `.env` file:
+```bash
+ENDPOINT="http://127.0.0.1:8545"
+PRIVATE_KEY="0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
+```
+## Grant role
+Next you need to grant `DEPLOYER_ROLE` for account that we have created in previous step for being able to deploy new contracts on chain. One of the easiest ways to do this would be to use the Blockscout interface. First you need to find the `ConfigController` contract, its address is `0xD200200000000000000000000000000000000d2`, click `Write Proxy` connect the wallet on which there is sFuel, find the `addToWhitelist` function and enter the address of the new account that you generated in the metamask, then click Write and confirm the transaction in the metamask.
+## Get sFuel
```
Comment (payvint) [inline proxy/scripts/upgrade_instruction.md]: You can add get sFUEL from Etherbase or from faucet available on your chain - look on sfuel.skale.network

**Review by payvint — APPROVED** [review]

**Pull request merged**